components/script/dom/mutationobserver.rs

```diff
@@ -152,21 +153,25 @@ impl MutationObserver {
         let mut i = 0;
         for observer in &interestedObservers {
             //Step 4.1
-            let mut record = MutationRecord::new(DOMString::from(&*given_name), target);
+            let mut record = Root::from_ref(&MutationRecord::new(DOMString::from(&*given_name), target));
             //Step 4.2
             if given_name != "" && given_namespace != "" {
-                record.SetAttributeName(DOMString::from(&*given_name));
-                record.SetAttributeNamespace(DOMString::from(&*given_namespace));
+//                mutableRootedRecord.deref().SetAttributeName(DOMString::from(&*given_name));
+//                mutableRootedRecord.deref().SetAttributeNamespace(DOMString::from(&*given_namespace));
             }
             //Step 4.3-4.6- TODO currently not relevant to mutation records for attribute mutations
             //Step 4.7
-            if pairedStrings[i] != ""{
-                record.SetoldValue(pairedStrings[i].clone());
+            if pairedStrings[i] != "" {
+//                mutableRootedRecord.deref().SetoldValue(pairedStrings[i].clone());
             }
             i = i + 1;
         }
         // Step 5
-        let enqueuedMutationCallback = EnqueuedMutationCallback { pipeline: target.global().pipeline_id() };
+        let cx: *mut JSContext = JSContext::new();
+        let callback: *mut JSObject = JSObject::new();
+        let mutationCallback: Rc<MutationCallback> = MutationCallback::new(cx, callback);
```

Comment:
Hi Josh, 

I have changed this by including a `&self` in the `queue_a_mutation_record` method's arguments and then calling it as:
`let enqueuedMutationCallback = EnqueuedMutationCallback { callback: self.callback, pipeline: target.global().pipeline_id() };`

but this gives the following error:
```
error[E0507]: cannot move out of borrowed content
   --> /home/sumit/Dropbox/NCSU/OODD/FinalProject/servo/components/script/dom/mutationobserver.rs:170:77
    |
170 |         let enqueuedMutationCallback = EnqueuedMutationCallback { callback: self.callback,
    |                                                                             ^^^^ cannot move out of borrowed content
```
Also, the same error is there for `microtask.rs` (line 92) in the following line:
`let observer = MutationObserver::Constructor(&global, job.callback);`
error:
```
error[E0507]: cannot move out of borrowed content
  --> /home/sumit/Dropbox/NCSU/OODD/FinalProject/servo/components/script/microtask.rs:92:83
   |
92 |                             let observer = MutationObserver::Constructor(&global, job.callback);
   |                                                                                   ^^^ cannot move out of borrowed content
```
Can you please help me out with this error?